### ISSUE TITLE
Search API results should return consist language-field formats (3-letter language-codes)

### ISSUE THREAD

**Issue by waldenn** [internetarchive/iaux#105]
**Description**
> What problem are we solving? What does the experience look like today? What are the symptoms? Please describe the component/story/add on related to the issue.

The search API results currently are not consistent wrt the language-field format.

See this [example search](https://archive.org/advancedsearch.php?q=odense&fl[]=language&sort[]=&sort[]=&sort[]=&rows=50&page=1&output=json&save=yes)

English items are returned both as " "english" and as "eng" and the same is true for other languages.

**Evidence / Screenshot (if possible)**

See example link above.

**Expectation**
> What should by happening? What will it look like / how will it behave?

There should be one language-field format used for all languages, consisting of the [3-letter-code](https://www.loc.gov/standards/iso639-2/php/code_list.php) of that language.

**Details**
> Browser type and version?

All.

> Operating system?

All.

**Proposal & Constraints**

> What is the proposed solution / implementation? Is there a precedent of this approach succeeding elsewhere?

Convert every non-3-letter language string in the database into its corresponding 3-letter-language string. Any new item inserted should conform to this 3-letter-code.

**Stakeholders**
> @ tag stakeholders of this bug

I hope this repo is the correct place to report this bug, else please let me know. Thanks for an awesome project!

**Comment by mitra42** [comment]
There are two parts to solving this .
A - the work that Arthur is doing (he's not in this repo) to clean up the metadata, i don't know if this difference is on his list
B - The work in the [dweb-archive](https://github.com/@Internetarchive/dweb-archive) and  [dweb-archivecontroller](https://github.com/@Internetarchive/dweb-archivecontroller) to clean up the results of the metadata API call to enforce contracts (for example enforcing that metadata items supposed to be singles are turned into strings and those that can be multiple are turned into arrays. In my opinion that would also be the right place to fix data problems like this until Arthur can get to them.  Note archivecontroller/Util.js contains a mapping between the three letter string and the full one for languages that I pulled out of the PHP code.  

Note this code is gradually moving into IAUX , but since its just me the process is relatively slow, but if you were thinking of rewriting this thn it might be better to start with code I've already built.

**Comment by waldenn** [comment]
Hi, thanks. The link to Util.js is appreciated for confirmation. I had already added most of the 3-letter codes to my Wikipedia-table (which uses 2-letter codes) in my project: https://wikischool.org/explore
I can now query Archive.org & Wikipedia for a specific language the user has selected.

Somewhat unrelated: Why not always using an array for data fields that can contain multiple values? Personally I think this makes the API simpler to use. Now every API user first needs to check the datatype before use.

**Comment by mekarpeles** [comment]
> Also: would it not be better to have a separate github project / issue-list for the Archive.org API interfaces?

You're welcome to post archive.org API comments here: https://github.com/archivelabs/archive.org/issues

Right now we don't use github for the archive.org backend + APIs.

**Comment by mitra42** [comment]
@waldenn - yes, it would be great if the metadata JSON API was consistent - either always arrays, or always arrays for any field that can be muli-value and always strings for the rest.   Most people who build on the API's seem to do this fix first, since otherwise every time you use any value from the API you have to test it.   

I have a document of many of the inconsistencies and weirdnesses in our API's but its not public yet.

**Comment by mitra42** [comment]
Also @waldenn  not sure if you are thanking me for the link to Util.js or asking for it, its in the dweb-archivecontroller repo I linked to, at the top level.

**Comment by waldenn** [comment]
Yes, I was just thanking for all the good work you people are doing! Looking forward to seeing that list : )
